executor/cook/executor.py

```diff
@@ -0,0 +1,115 @@
+"""
+An implementation of a Mesos executor, using the agent HTTP API via pymesos.
+
+It's primary responsibility is responding to Mesos callbacks and updating the status of the
+running task.
+"""
+
+import json
+import time
+import uuid
+import logging
+
+from pymesos import MesosExecutorDriver, Executor, encode_data, decode_data
+
+from cook.store import WATCH_ACTION_PUT
+
+def new_task_info(id, state, data = None):
+    """
+    Create a new TaskInfo dict.
+    """
+    return {
+        'state': state,
+        'task_id': {'value': id},
+        # FIXME: setting uuid manually is a workaround for a bug in pymesos
+        # https://github.com/douban/pymesos/issues/33
+        'uuid': encode_data(uuid.uuid4().bytes).decode('utf8'),
+        'data': encode_data(json.dumps(data).encode('utf8')).decode('utf8') if data else ''
+    }
+
+class CookExecutor(Executor):
+    def __init__(self, store, event, server, sandbox):
+        self.store = store
+        self.event = event
+        self.server = server
+        self.sandbox = sandbox
+
+    def registered(self, driver, executorInfo, frameworkInfo, slaveInfo):
+        """
+        Called when the executor is registered with the Mesos agent.
+
+        Adds a watch to store in order to keep task info in sync.
+        """
+        logging.info("CookExecutor:registered")
+
+        def sync_task_status(action, type, id, entity):
+            if action is WATCH_ACTION_PUT and type is 'task':
+                state = entity.get('state')
+
+                entity.update({'sandbox': self.sandbox})
+
+                if state:
+                    if state == 'TASK_FAILED' or state == 'TASK_FINISHED':
+                        self.server.stop()
```

Comment:
Nitpick: Should we add logs that server is being stopped as the task finished/failed?